crates/nu-cli/src/eval_cmds.rs

```diff
@@ -41,43 +38,33 @@ pub fn evaluate_commands(
 
         if let Some(err) = working_set.parse_errors.first() {
             report_error(&working_set, err);
-
             std::process::exit(1);
         }
 
         (output, working_set.render())
     };
 
     // Update permanent state
-    if let Err(err) = engine_state.merge_delta(delta) {
-        let working_set = StateWorkingSet::new(engine_state);
-        report_error(&working_set, &err);
-    }
+    engine_state.merge_delta(delta)?;
```

Comment:
It changes the original logic a little bit.  I don't think it's right to return early on error.

Reply:
It might I guess. I'm just not sure which behavior is correct. In the old `eval_cmds` we would report the error and continue. But in the old `eval_file` we would panic on `.expect("merging delta into engine_state should succeed")`. I changed both to just return the error instead. Should I change both of them back?

Reply:
I don't think need to change `eval_file.rs` back, because return `Error` is better than `expect`, and the running progress is the same: `merge_delta` failed, and return back.

But I'm not sure why nushell reports error in `eval_cmds.rs` either, before we change it return on error, maybe it's better to investigate when `merge_delta` failed.

Reply:
I looks like `merge_delta` can only fail if writing updates to the plugin file fails. I think this should be an error, as otherwise some plugin commands won't be persisted. In the future, I think we can change `merge_delta` to not return an error, since it looks the new `plugin add` and `plugin use` have separate handling, and the plugin code in `merge_delta` is only for `register`.

Reply:
OK, make sense to me :-)